onnxruntime/core/providers/coreml/builders/impl/gather_op_builder.cc

```diff
@@ -28,13 +30,37 @@ int64_t GetAxisAttribute(const Node& node) {
 }  // namespace
 
 Status GatherOpBuilder::AddToModelBuilderImpl(ModelBuilder& model_builder, const Node& node,
-                                              const logging::Logger& /*logger*/) const {
-  auto layer = model_builder.CreateNNLayer(node);
-  layer->mutable_gather()->set_axis(GetAxisAttribute(node));
-  *layer->mutable_input()->Add() = node.InputDefs()[0]->Name();    // data
-  *layer->mutable_input()->Add() = node.InputDefs()[1]->Name();    // indices
-  *layer->mutable_output()->Add() = node.OutputDefs()[0]->Name();  // output
-  model_builder.AddLayer(std::move(layer));
+                                              const logging::Logger& logger) const {
+  if (model_builder.CreateMLProgram()) {
+    using namespace CoreML::Specification::MILSpec;  // NOLINT
+    std::unique_ptr<Operation> op = model_builder.CreateOperation(node, "gather");
```

Comment:
if it's just for referring to one or a few types, can use `using <qualified type name>;`. that's generally preferable to `using namespace x;` which pulls all names into scope and increases the chance for conflicts.
```suggestion
    using CoreML::Specification::MILSpec::Operation;
    std::unique_ptr<Operation> op = model_builder.CreateOperation(node, "gather");
```